zcash_client_backend/src/fees/common.rs

```diff
@@ -533,10 +538,10 @@ where
                         let fee_with_dust = (total_in - total_out)
                             .expect("we already checked for sufficient funds");
```

Comment:
`total_out <= total_out_plus_fee`, so this is correct once we've added back the check at line 480. However, since `total_change == total_in - (total_out + total_fee)`, we can also calculate it like this:
```suggestion
                        let fee_with_dust = (total_change + total_fee).ok_or_else(overflow)?
```
This is better because it makes the assertion redundant.